src/js/iterator.js

```diff
@@ -234,6 +244,7 @@ export default class Iterator {
     while (true) {
       let comparison = compare(position, this.currentNodePosition)
 
+      this.cacheNodePosition()
```

Comment:
Yeah, I was going to put it somewhere more central originally, but thought that I'd rather avoid the overhead of doing the node caching during traversal while inserting or splicing.

I didn't try benchmarking that other approach though, so maybe we should refactor to something like that if we find that it's easier to maintain and doesn't slow things down.